istio/discovery.go

```diff
@@ -754,11 +777,25 @@ func (in *Discovery) Mesh(ctx context.Context) (*models.Mesh, error) {
 		}
 	}
 
+	// add set the NamespaceMap
+	mesh.NamespaceMap = map[string]*models.ControlPlane{}
+	for _, cp := range controlPlanes {
+		for _, mn := range cp.ManagedNamespaces {
+			key := Key(mn.Cluster, mn.Name)
```

Comment:
`Key` should not be exported otherwise something else will end up using this and it probably shouldn't. Same for `NamespaceMap`. `namespaceMap` should be a field on `discovery` instead?

Reply:
fixed.